sources/packages/backend/apps/api/src/services/education-program/education-program.service.ts

```diff
@@ -227,7 +240,16 @@ export class EducationProgramService extends RecordDataModelService<EducationPro
     } else {
       program.modifier = auditUser;
     }
-    return this.repo.save(program);
+    return this.dataSource.transaction(async (transactionalEntityManager) => {
+      await this.saveInstitutionAddsPendingProgramNotification(
+        { name: program.name, programStatus: program.programStatus },
+        institutionId,
+        transactionalEntityManager,
+      );
+      return transactionalEntityManager
```

Comment:
https://github.com/bcgov/SIMS/pull/3122#discussion_r1575462371

Reply:
I didn't find a `try/catch` block here. So, didn't add the `await` with the `return`.